### PR TITLE
Feat/search

### PR DESCRIPTION
**Pull request by Khilio1081**
**OVERVIEW**
Added Navigation bar

**SNIPPETS**
image

### CODE REVIEW

File: src/app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class UserController extends Controller
+{
+    protected $userService;
+
+    public function __construct(UserService $userService)
+    {
+        $this->userService = $userService;
+    }
+
+    public function searchUsernames(Request $request)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: docbloc comment and return type

File: src/app/Services/UserService.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\User;
+
+class UserService
+{
+    public function searchUsernames(string $query)
```
Comment (hipe01mark) [inline src/app/Services/UserService.php]: comment and return type

File: src/public/js/charcount.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    if (characterCount >= 100) {
+        maximum.css('color', '#8f0001');
+        current.css('color', '#8f0001');
+        theCount.css('font-weight', 'strong');
+    } else {
+        maximum.css('color', '#666');
+        theCount.css('font-weight', 'normal');
+    }
```
Comment (hipe01mark) [inline src/public/js/charcount.js:28]: prevent else in the if statement

File: src/public/js/charcount.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+$('textarea').keyup(function () {
+
+    var characterCount = $(this).val().length,
+        current = $('#current'),
+        maximum = $('#maximum'),
+        theCount = $('#the-count');
```
Comment (hipe01mark) [inline src/public/js/charcount.js:6]: use const

File: src/public/js/editProfile.js [hunk 5]
```diff
@@ -1,19 +1,19 @@
-function showDialog() {
-    let dialog = document.getElementById("dialog");
-    dialog.classList.remove("hidden");
-    dialog.classList.add("flex")
+function showProfile() {
+    let profile = document.getElementById("profile");
```
Comment (hipe01mark) [inline src/public/js/editProfile.js:2]: use const

File: src/public/js/editProfile.js [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
+        profile.classList.add("opacity-100")
     }, 20);
 }
 
-function hideDialog() {
-    let dialog = document.getElementById("dialog");
-    dialog.classList.add("opacity-0")
-    dialog.classList.remove("opacity-100")
+function hideProfile() {
+    let profile = document.getElementById("profile");
```
Comment (hipe01mark) [inline src/public/js/editProfile.js:11]: use const

File: src/public/js/logout.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+document.getElementById('profile-image-container').addEventListener('click', function() {
+    var dropdownMenu = document.getElementById('dropdown-menu');
```
Comment (hipe01mark) [inline src/public/js/logout.js]: use const

File: src/public/js/logout.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    dropdownMenu.classList.toggle('hidden');
+    dropdownMenu.classList.toggle('show');
+});
+
+document.addEventListener('click', function(event) {
+    var profileImageContainer = document.getElementById('profile-image-container');
+    var dropdownMenu = document.getElementById('dropdown-menu');
+
+    var isClickInside = profileImageContainer.contains(event.target) ||
+                        dropdownMenu.contains(event.target);
```
Comment (hipe01mark) [inline src/public/js/logout.js]: use const

File: src/public/js/charcount.js (hunk 3 above)
Comment (Khilio1081) [reply src/public/js/charcount.js:28]: Resolved by Paul

File: src/public/js/charcount.js (hunk 4 above)
Comment (Khilio1081) [reply src/public/js/charcount.js:6]: Resolved by Paul

File: src/public/js/editProfile.js (hunk 5 above)
Comment (Khilio1081) [reply src/public/js/editProfile.js:2]: Resolved by Paul

File: src/public/js/editProfile.js (hunk 6 above)
Comment (Khilio1081) [reply src/public/js/editProfile.js:11]: Resolved by Paul

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request closed without merging**